### PR TITLE
build contract staking indexer only when staking protocol enabled

### PR DESCRIPTION
**Pull request by envestcc**
# Description
Currently, the usage of the contract stake indexer is limited to the staking module. Therefore, if the staking protocol is not enabled, there is no need to create the indexer.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: chainservice/builder.go [hunk 1]
```diff
@@ -298,6 +298,9 @@ func (builder *Builder) buildSGDRegistry(forTest bool) error {
 }
 
 func (builder *Builder) buildContractStakingIndexer(forTest bool) error {
+	if !builder.cfg.Chain.EnableStakingProtocol {
```
Comment (Liuhaai) [inline chainservice/builder.go:301]: will `contractStakingIndexer` be `nil` when enabled?

File: chainservice/builder.go [hunk 2]
```diff
@@ -298,6 +298,9 @@ func (builder *Builder) buildSGDRegistry(forTest bool) error {
...
+		return nil
+	}
```
Comment (dustinxie) [inline chainservice/builder.go:303]: it seems this flag is defined but not used at all? we should remove it then

**Review by dustinxie — APPROVED** [review]

File: chainservice/builder.go (hunk 1 above)
Comment (dustinxie) [reply chainservice/builder.go:301]: it could be, if `cfg.Genesis.SystemStakingContractAddress == ""` in the config

Comment (Liuhaai) [reply chainservice/builder.go:301]: should ContractStakingIndexer be built when cfg.Genesis.SystemStakingContractAddress == ""

Comment (dustinxie) [reply chainservice/builder.go:301]: no, in that case, means SystemStaking (IIP-13) is not enabled on the chain, no need to build the indexer

**Pull request merged**